startupscript/git-setup.sh

```diff
@@ -46,5 +46,22 @@ ${RUN_AS_LOGIN_USER} "mkdir -p '${WORKBENCH_GIT_REPOS_DIR}'"
 # to the git references, the corresponding git repo cloning will be skipped.
 # Keep this as last thing in script. There will be integration test for git cloning (PF-1660). If this is last thing, then
 # integration test will ensure that everything in script worked.
-${RUN_AS_LOGIN_USER} "cd '${WORKBENCH_GIT_REPOS_DIR}' && wb git clone --all"
 
+# This loop replaces the logic of "wb git clone --all", which currently does not work without
+# Google Application Default Credentials being available.  This emulates the behavior of the CLI
+# command, continuing with an error message when an individual repo cannot be cloned.
+
+# shellcheck disable=SC2164
+pushd "${WORKBENCH_GIT_REPOS_DIR}"
+${RUN_AS_LOGIN_USER} "wb resource list --type=GIT_REPO --format json" | \
+  jq -c .[] | \
+  while read -r ITEM; do
```

Comment:
There's really (IMO) nasty feature of Bash whereby:

```
<cmd> | while read ...; do
           <cmds>
        done
```

creates a subshell for the cmds to run in. Bash does not exit the out script with failure in the subshell.
You observed this in that a "git clone" failures do not exit this `while ... do` nor the script as a whole.

https://lists.gnu.org/archive/html/bug-bash/2008-01/msg00121.html

There are better patterns to follow as described here in the Google shell style guide:

https://lists.gnu.org/archive/html/bug-bash/2008-01/msg00121.html

We should *consider* enforcing the use of these other patterns to avoid the unexpected error behavior.

Reply:
Just to clarify my previous comment, I did observe the undesired behavior (exit script on failed clone) before I added the `|| echo "git clone of ${GIT_REPO_URL} failed.""` to the `git clone` line inside the loop to allow it to continue.

Reply:
Interesting! I'll see if I can surface more info later, but I'm glad to see that (and I just did some testing that allays my fears here).

The issue of variable changes not propagating outward due to the subshell is real, and we should be aware of that, but this doesn't appear to be an immediate concern.